src/Countries/Pakistan.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Spatie\Holidays\Countries;
+
+use Carbon\CarbonImmutable;
+
+class Pakistan extends Country
+{
+    public function countryCode(): string
+    {
+        return 'pk';
+    }
+
+    protected function allHolidays(int $year): array
+    {
+        return array_merge([
+            'Kashmir Solidarity Day' => '02-05',
+            'Pakistan Day' => '03-23',
+            'Labour Day' => '05-01',
+            'Independence Day' => '08-14',
+            'Iqbal Day' => '11-09',
+            'Quaid-e-Azam Day' => '12-25',
+        ], $this->variableHolidays($year));
+    }
+
+    /** @return array<string, CarbonImmutable> */
+    protected function variableHolidays(int $year): array
+    {
+        // The variable holidays are all follow lunar calendar so their dates are not confirmed.
```

Comment:
You can look at how Turkey solved this:
https://github.com/spatie/holidays/blob/main/src/Countries/Turkey.php#L11

Reply:
I checked the implementation, the solution provides the dates but there is no surety that the calendar will follow thsoe exact dates. Should I go for it anyway?

Reply:
@Nielsvanpach I tried adapting the solution. The problem is, that we have 4 different sets of holidays that follow the lunar calendar(total of 10 holidays) and it is making the code a lot more complex and even after that, the dates will not be accurate.